Add cross environment configuration

Atvaark opened the pull request:
This optionally allows users to configure variables and config changes
in the 'default' environment in addition to concrete environments
(e. g. Dev, Prod). The changes defined in the 'default' environemnt
will be applied before the changes defined in the concrete ones.
This allows overwriting default changes with changes that are
environment specific.

Using just a single 'default' environment to insert tokens and
configuring the token values via TFS variables will be possible
with this change.

Compared to #17 this also allows defining cross-environment CustomVariables.

A review and tests (especially for the PowerShell 3 script) would be appreciated, as I couldn't get the tokenizer unit tests to run.

harshil93 commented:
@Atvaark Can you also add some test cases around this

Atvaark commented:
I added some unit tests for both default variables and config changes.

harshil93 commented:
@Atvaark You will need to bump up the minor version in task.json also.

harshil93 commented on Utilites/Tokenizer/tokenize-ps3.ps1:
```diff
@@ -107,11 +113,12 @@ try {
-                if ($Configuration.$environmentName.CustomVariables.$matchedItem) {
-                    $variableValue = $Configuration.$environmentName.CustomVariables.$matchedItem
-                    Write-Verbose "Found variable '$matchedItem' in configuration with value '$variableValue" 
+                $environmentVariableValue = $environmentNames | foreach { $Configuration.$_.CustomVariables.$matchedItem } | where { $_ } | select -Last 1
```
Can you add a comment here which says that select -Last 1 is the logic which selects the right value since all the overriding environments should be later in the $environmentNames list than default env.

Atvaark commented:
* The task version is now 2.1.0
* Added a comment to the line that selects the env-value or falls back to the default value
* Added a description to the overview.md
* Fixed some markdown issues (GitHub updated their Markdown grammar a few months ago)
